spec/controllers/your_results_controller_spec.rb

```diff
@@ -15,6 +15,17 @@
   let(:period_presenters) { [period_presenter, period_presenter] }
   let(:period_presenter) { double(Wpcc::PeriodPresenter) }
 
+  describe 'GET /your_results' do
+    context 'when session has no keys' do
+      it 'redirects to your details step' do
```

Comment:
@tomas-stefano should this say `it 'redirects to root page' do`? 

I know that the your details step is, at present, the root path. However, unlikely it may sound, it is possible that another page can be made to become a landing page. In view of that I believe that the test should expect the redirect to go to the your details path.

```
describe 'GET /your_results' do
  context 'when session has no keys' do
    it 'redirects to your details step' do
      get :index, {}, {}

      expect(response).to redirect_to new_your_detail_path(locale: :en)
    end
  end
end
```

I tried the solution above and here is the result

```
[jonesagyemang:~/Projects … wpcc] 8448-session-expired-redirect(+1/-1)+* ± rspec spec/controllers/your_results_controller_spec.rb
F....

Failures:

  1) Wpcc::YourResultsController GET /your_results when session has no keys redirects to your details step
     Failure/Error:
       expect(response)
         .to redirect_to new_your_detail_path(locale: :en)

       Expected response to be a redirect to <http://test.host/en/tools/workplace-pension-contribution-calculator/your_details/new> but was a redirect to <http://test.host/en/tools/workplace-pension-contribution-calculator/>.
       Expected "http://test.host/en/tools/workplace-pension-contribution-calculator/your_details/new" to be === "http://test.host/en/tools/workplace-pension-contribution-calculator/".
     # ./spec/controllers/your_results_controller_spec.rb:23:in `block (4 levels) in <top (required)>'

Finished in 0.08779 seconds (files took 1.7 seconds to load)
5 examples, 1 failure

Failed examples:

rspec ./spec/controllers/your_results_controller_spec.rb:20 # Wpcc::YourResultsController GET /your_results when session has no keys redirects to your details step

Coverage report generated for RSpec to /Users/jonesagyemang/Projects/MAS/wpcc/coverage. 206 / 259 LOC (79.54%) covered.
```

Clearly that doesn't work. Therefore, you could think about renaming the it _step_ to reflect that the intention is to drive the user back to the root page rather than being specific about what that page is -- `it 'redirects to root page' do`